### ISSUE TITLE
Should be spacing around = in for loop initializer

### ISSUE THREAD

**Issue by fangism** [google/verible#507]
**Test case**

```systemverilog
function void f();
for (int i=N; i > 0; i--) begin
end
endfunction
```

**Actual output**

```systemverilog
function void f();
  for (int i=N; i > 0; i--) begin
end
endfunction
```

**Expected or suggested output**

```systemverilog
function void f();
  for (int i = N; i > 0; i--) begin
  end
endfunction
```

Should be `i = N`.  `end` indentation looks wrong, and will be separate issue.

**Comment by fangism** [comment]
`end` indentation is #508

**Comment by fangism** [comment]
b/169606046

**Comment by fangism** [comment]
Checking spacing annotation with `-v=4 --alsologtostderr`, the spacing constraint looks correct:

```
I0929 15:55:32.545721 1416352 token_annotator.cc:136] Spacing between SymbolIdentifier and '='
I0929 15:55:32.545723 1416352 token_annotator.cc:554] spaces: 1, reason: Space around binary and assignment operators
I0929 15:55:32.545725 1416352 token_annotator.cc:677] Inter-token penalty between SymbolIdentifier and '='
I0929 15:55:32.545733 1416352 token_annotator.cc:682] context break penalty: 18
I0929 15:55:32.545735 1416352 token_annotator.cc:686] inter-token break penalty: 5, right is '='
I0929 15:55:32.545738 1416352 token_annotator.cc:691] token+context break penalty: 0, No adjustment.
I0929 15:55:32.545740 1416352 token_annotator.cc:701] total break penalty: 28
I0929 15:55:32.545742 1416352 token_annotator.cc:895] line break constraint: undecided: Default: leave wrap decision to algorithm
I0929 15:55:32.545744 1416352 token_annotator.cc:136] Spacing between '=' and SymbolIdentifier
I0929 15:55:32.545746 1416352 token_annotator.cc:554] spaces: 1, reason: Space around binary and assignment operators
I0929 15:55:32.545748 1416352 token_annotator.cc:677] Inter-token penalty between '=' and SymbolIdentifier
```

But it looks like a range of text was inadvertently format disabled (preserving).

```
I0929 15:56:30.669087 1416352 formatter.cc:382] Does not fit, preserving.
I0929 15:56:30.669090 1416352 format_token.cc:255] PreserveSpacesOnDisabledTokenRanges
I0929 15:56:30.669093 1416352 format_token.cc:262] disabling bytes: [20, 54)
I0929 15:56:30.669097 1416352 format_token.cc:267] disabling tokens: [7, 22)
I0929 15:56:30.669100 1416352 format_token.cc:277] checking whether first ftoken in range is a must-wrap.
I0929 15:56:30.669103 1416352 format_token.cc:291] disable-format preserve spaces before: (#40: "(")
I0929 15:56:30.669106 1416352 format_token.cc:291] disable-format preserve spaces before: (#506: "int")
I0929 15:56:30.669109 1416352 format_token.cc:291] disable-format preserve spaces before: (#293: "i")
I0929 15:56:30.669112 1416352 format_token.cc:291] disable-format preserve spaces before: (#61: "=")
I0929 15:56:30.669114 1416352 format_token.cc:291] disable-format preserve spaces before: (#293: "N")
I0929 15:56:30.669122 1416352 format_token.cc:291] disable-format preserve spaces before: (#59: ";")
I0929 15:56:30.669125 1416352 format_token.cc:291] disable-format preserve spaces before: (#293: "i")
I0929 15:56:30.669128 1416352 format_token.cc:291] disable-format preserve spaces before: (#62: ">")
I0929 15:56:30.669131 1416352 format_token.cc:291] disable-format preserve spaces before: (#300: "0")
I0929 15:56:30.669134 1416352 format_token.cc:291] disable-format preserve spaces before: (#59: ";")
I0929 15:56:30.669136 1416352 format_token.cc:291] disable-format preserve spaces before: (#293: "i")
I0929 15:56:30.669139 1416352 format_token.cc:291] disable-format preserve spaces before: (#696: "--")
I0929 15:56:30.669142 1416352 format_token.cc:291] disable-format preserve spaces before: (#41: ")")
I0929 15:56:30.669154 1416352 format_token.cc:291] disable-format preserve spaces before: (#319: "begin")
I0929 15:56:30.669158 1416352 format_token.cc:291] disable-format preserve spaces before: (#333: "end")
```

which seems related to #508.

**Comment by fangism** [comment]
Confirmed, the fix for #508 fixed this one too, but I'll add/modify a test case to cover this example.
